addons/product/static/src/js/product_template_import.js

```diff
@@ -1,19 +1,75 @@
-odoo.define('product_template_import.import', function (require) {
+odoo.define('product.import_template', function (require) {
 "use strict";
 
-var core = require('web.core');
 var BaseImport = require('base_import.import');
+var core = require('web.core');
+var ImportTemplatesRegistry = require('base_import.import_templates_registry');
 
 var _t = core._t;
 
+ImportTemplatesRegistry.add('product.supplierinfo', {
+    text: _t('Import Template for Vendor Pricelists'),
+    template: '/product/static/xls/product_supplierinfo.xls'
+});
+ImportTemplatesRegistry.add('product.pricelist', {
+    text: _t('Import Template for Pricelists'),
+    template: '/product/static/xls/product_pricelist.xls'
+});
+
 BaseImport.DataImport.include({
-    renderImportLink: function() {
-        this._super();
-        if (this.res_model == 'product.template') {
-            this.$(".import-link").prop({"text": _t(" Import Template for Products"), "href": "/product/static/csv/product.template.csv"});
-            this.$(".template-import").removeClass("hidden");
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Set template link based on some dynamic cases:
+     * - If pricelist not activated or pricelist activated with formulas
+     * - If default option of pricelist is activated (several prices on products)
+     * - If purchase module then set different template for products
+     *
+     * @override
+     */
+    renderImportLink: function () {
```

Comment:
The entire selection of import templates should be moved to the server side and properly overloaded on a per-model basis. The JS side should live only in the base import module and just call out over RPC to get the relevant link & label.